### PR TITLE
+str #19069 Add FlowOp.prepend for prepending Sources to Flows

### PR DESCRIPTION
**Pull request by reavowed**
See #19069

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala [hunk 1]
```diff
@@ -909,6 +909,42 @@ final class Flow[-In, +Out, +Mat](delegate: scaladsl.Flow[In, Out, Mat]) extends
...
+   * '''Emits when''' element is available from the given [[Source]] or from current stream when the [[Source]] is completed
+   *
+   * '''Backpressures when''' downstream backpressures
+   *
+   * '''Completes when''' this [[Flow]] completes
+   *
+   * '''Cancels when''' downstream cancels
+   */
+  def prepend[T >: Out, M](that: Graph[SourceShape[T], M]): javadsl.Flow[In, T, Mat] =
+    new Flow(delegate.prepend(that))
```
Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala]: Naming wise wouldn't `from` be the nicer opposite to `to` which we use for `flow to sink`?

**Comment by ktoso** [comment]
OK TO TEST

File: akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala (hunk 1 above)
Comment (reavowed) [inline akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala]: This is an opposite to `concat`, not `to` (it produces a Flow of the same type, not a Sink).

Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala]: Ah, right; should have read the scaladoc. :+1:

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowConcatSpec.scala [hunk 2]
```diff
@@ -40,6 +40,23 @@ class FlowConcatSpec extends BaseTwoStreamsSetup {
...
+      val subSink = Sink.publisher[Any](false)
+
+      val (_, res) = f2.prepend(s1).runWith(s2, subSink)
+
+      res.subscribe(subs)
+      val sub = subs.expectSubscription()
+      sub.request(9)
+      (1 to 6).foreach(e ⇒ subs.expectNext(e.toString + "-s"))
+      subs.expectComplete()
+    }
```
Comment (rkuhn) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowConcatSpec.scala]: Please also add to SourceTest and FlowTest on the Java side (just to verify that the API can be invoked, high-level).

**Comment by rkuhn** [comment]
Please also add to SubSource and SubFlow on the Java side, and add to the operations overview listing in the reference docs (PR validation was actually a success, but it would fail now due to DslConsistencySpec).

**Comment by rkuhn** [comment]
And thanks for the contribution!

**Comment by ktoso** [comment]
Hi @reavowed, do you think you could have a look at the suggestions for this PR @rkuhn had?
It's a nice addition and we'd like to merge it once completed :)

**Comment by reavowed** [comment]
Sure, was planning to get to it later today :)

**Comment by ktoso** [comment]
Great, thanks in advance!

**Comment by rkuhn** [comment]
OK TO TEST

**Comment by rkuhn** [comment]
You’ll need to add `"prependGraph"` to the graph helpers list in DslConsistencySpec, then it’ll be all green.

**Comment by rkuhn** [comment]
fixing up and merging manually

**Comment by rkuhn** [comment]
merged

**Pull request closed without merging**